lib/pulse/lib/Pulse.Lib.ArrayPtr.fst

```diff
@@ -17,22 +17,108 @@
 module Pulse.Lib.ArrayPtr
 #lang-pulse
 
+noeq
+type base_t ([@@@strictly_positive] a: Type0) =
+| Null
+| Ref of R.ref a
+| Array of A.array a
```

Comment:
FYI, these days `array a` has both null pointers (`A.null`) and you can convert references to arrays (`R.to_array`).

Reply:
Thanks Gabriel! I reimplemented `from_ref`, etc. accordingly.